client/src/components/SponsorsComps/SponserGeneralComps.jsx

```diff
@@ -0,0 +1,142 @@
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  margin: ${space.xl} ${space.s};
+`}
+`;
+
+const ImageContainer= styled.a`
+  ${( { theme: { color, space } } ) => `
+      width: 248px;
+      height: 230px;
+      max-width: 248px;
+
+      position: relative;
+      display: flex;
+      justify-content: center;
+      text-align: center;
+      align-items: center;
+      margin-bottom: ${space.m};
+
+      color: inherit;      
+      border: 4px solid ${color.box_shadow_1};
+      background: ${color.image_link_bkgr};
+      text-decoration: none;
+      cursor: pointer;
+      transition: opacity 300ms linear;
+      
+      &:hover{
+        opacity: 0.5;
+      }
+    `}
+`;
+
+const SponsorImg = styled.img`
+    ${ ({ theme: { space } }) => `
+      max-height: 240px;
+      max-width: 240px;
+      padding: ${space.s};
+    `}
+`;
+
+const SMPLinkingButton = styled(ButtonStyledLink)`
+  &:hover{
+    color: ${props => props.theme.color.text_1};
+  }
+`;
+
+const FAButtonContainer = styled.div`
+  display: flex;
+`
+
+const FAButton = styled.button`
+${ ({ theme: { color } }) => `
+  width: 50%;
+  height: 40px;
+  
+  display: flex;
+  justify-content: center;
+  
+  user-select: none;
+  cursor: pointer;
+  
+  color: ${color.text_1};
+  text-decoration: none;
+  background: linear-gradient(to right, ${color.font_awsome_background_1} 0%, ${color.font_awsome_background_2} 100%);
+  border: 0;
+  border-radius: 0;
+  border-color: ${color.font_awsome_border};
+  transition: all 200ms;
+  
+  &:hover{
+    color: ${color.text_1};
+    text-decoration: none;
+    background: linear-gradient(to right, ${color.font_awsome_background_1} 0%, ${color.font_awsome_background_2} 100%);
+    box-shadow: inset 0px 0px 10px 2px ${color.font_awsome_box_shadow_1}, 0px 0px 10px 0px ${color.font_awsome_box_shadow_2};
+  }
+`}
+`;
+
+//React components section
+
+export class SponsorMiniPremium extends React.Component {
+  state = {
+    hovered: false
+  };
+  render() {
+    const { 
+      name,
+      logo,
+      onEdit,
+      onDelete,
+      isOnWhite
+    } = this.props;
+
+    return (
+      <div>
+        {onEdit && (
+          <FAButtonContainer>
+            <FAButton 
+              onClick={onEdit}
+            >
+              <FontAwesomeIcon icon={faPencilAlt} />
+            </FAButton>
+            <FAButton
+              onClick={onDelete}
+            >
+              <FontAwesomeIcon icon={faTrash} />
+            </FAButton>
+          </FAButtonContainer>
+        )}
+
+        <SponsorItem>
+            <ImageContainer href={`/sponsor/${name}`}>
+              <SponsorImg
+                src={image(logo, 350, 150)}
+                alt={name}
+              />
+            </ImageContainer>
+            <SMPLinkingButton href={`/sponsor/${name}`}>
+              Explore Opportunities
+            </SMPLinkingButton>
+
+        </SponsorItem>
+      </div>
+    );
+  }
+};
```

Comment:
I see there is a single export here.
Are e going to have more components in this `GeneralComps` ? 
If not, lets have a file per component, its easier to maintain.

Reply:
I'm still not sure. Let's wait with it till the end of the sponsors bit.